presto-hive/src/test/java/com/facebook/presto/hive/metastore/InMemoryHiveMetastore.java

```diff
@@ -162,22 +165,28 @@ public synchronized void alterDatabase(String databaseName, Database newDatabase
     @Override
     public synchronized void createTable(Table table)
     {
-        SchemaTableName schemaTableName = new SchemaTableName(table.getDbName(), table.getTableName());
-        Table tableCopy = table.deepCopy();
-        if (tableCopy.getSd() == null) {
-            tableCopy.setSd(new StorageDescriptor());
+        TableType tableType = TableType.valueOf(table.getTableType());
+        checkArgument(EnumSet.of(MANAGED_TABLE, EXTERNAL_TABLE, VIRTUAL_VIEW).contains(tableType), "Invalid table type: %s", tableType);
+
+        if (tableType == VIRTUAL_VIEW) {
+            checkArgument(table.getSd().getLocation() == null, "Storage location for view must be null");
         }
-        else if (tableCopy.getSd().getLocation() != null) {
-            File directory = new File(new Path(tableCopy.getSd().getLocation()).toUri());
+        else {
+            File directory = new File(new Path(table.getSd().getLocation()).toUri());
             checkArgument(directory.exists(), "Table directory does not exist");
-            checkArgument(isParentDir(directory, baseDirectory), "Table directory must be inside of the metastore base directory");
+            if (tableType == MANAGED_TABLE) {
+                checkArgument(isParentDir(directory, baseDirectory), "Table directory must be inside of the metastore base directory");
```

Comment:
This is in-memory for testing, so I don't think it matters. I assume this check is to (1) make sure the table location picking does the right thing and (2) we don't accidentally delete or overwrite the wrong file system path.